src/state.py

```diff
@@ -89,6 +92,11 @@ def from_charm(cls, charm: ops.CharmBase) -> "HealthCheck":
             if charm.config.get("health-check-port") is not None
             else None
         )
+        all_or_none_heatth_checks_set = bool(interval) == bool(rise) == bool(fall)
```

Comment:
Maybe we should do a pydantic validation instead of doing it here so that we clearly separate the responsibilities ( since like this we're doing both validation in the model with `Field` and in the `from_charm` method.